### PR TITLE
tgui strip panel + strippable element

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request

https://user-images.githubusercontent.com/35135081/111956242-c8ebcc00-8aa7-11eb-9034-26b3e5716a55.mp4

Completely removes `show_inv` and replaces it with `/datum/element/strippable`. It takes a list of instantiated `/datum/strippable_item` which communicate which slots are available and how to interact with them. This element has been added to humans, alien humanoids, parrots, and corgis.

Some stuff, either through process of refactoring or for the sake of better code elsewhere, has been removed:
- Previously, clicking on a corgi's "back" slot without an item in your hand would pet them. This has been removed, just pet them normally.
- Remove alien's `r_store` and `l_store`, which were said to be the pouches in `show_inv`. I saw no reference of them elsewhere, and so removed them.
- Previously, attaching C4 to a slot of a corgi would arm it and attach. This has been removed, as it is identical to just placing the C4 on the corgi.

Through process of refactoring, the wording of some stuff has changed as well (mostly changing generic text such as "you can't put that" to "you can't put [item]").

## Why It's Good For The Game
Looks prettier, and modularizes a crap ton of copy and paste code that exists now.

## Changelog
:cl:
refactor: The strip panel has been completely remodeled in tgui.
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/StripMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+    for (let columnIndex = 0; columnIndex < COLUMNS; columnIndex++) {
+      const keyAtSpot = gridSpots.get(getGridSpotKey([rowIndex, columnIndex]));
+      if (!keyAtSpot) {
+        buttons.push(
+          <Stack.Item
+            style={{
+              width: BUTTON_DIMENSIONS,
+              height: BUTTON_DIMENSIONS,
+            }}
+          />
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/StripMenu.tsx:267]: These things are missing the keys in all push calls, and ESLint doesn't catch this pattern.

Honestly, I advise against this imperative approach, I think you can totally do it declaratively.

**Comment by Sparkezel** [comment]
Did you test on monkies and Ian?

**Comment by Mothblocks** [comment]
Monkies are species of humans now, the only other carbons are aliens, which are tested.

As said in the post, corgis are supported, so yes.

The heretic cloak, as far as I know, is not invisible on the current menu either, so that is not a regression.

File: tgui/packages/tgui/interfaces/StripMenu.tsx (hunk 1 above)
Comment (Mothblocks) [reply tgui/packages/tgui/interfaces/StripMenu.tsx:267]: Interested to see how this'd be done less imperatively, how might that look like?

Comment (stylemistake) [reply tgui/packages/tgui/interfaces/StripMenu.tsx:267]: Yeah, I just mean map (over a range of numbers).

**Comment by Colovorat** [comment]
Maybe some action that takes times to inspect what in pockets of person? So sec will not remove tanks from plasmemes

**Comment by stylemistake** [comment]
shift click

**Comment by Twaticus** [comment]
does this also fix the issue of the stripping menu reopening after youd already closed it

**Comment by Mothblocks** [comment]
@Colovorat This is a refactor meant to emulate existing UX as much as possible, you are free to make your own PR after the fact since the code for this is very easy to update.

@Twaticus Yes.


https://user-images.githubusercontent.com/35135081/112053818-113fd400-8b12-11eb-909b-c89d93ead98a.mp4

File: tgui/packages/common/math.ts [hunk 2]
```diff
@@ -98,3 +98,10 @@ export const numberOfDecimalDigits = value => {
   }
   return 0;
 };
+
+/**
+ * Returns a range of numbers from start to end, exclusively.
+ * For example, range(0, 5) will return [0, 1, 2, 3, 4].
+ */
+export const range = (start: number, end: number): number[] =>
```
Comment (stylemistake) [inline tgui/packages/common/math.ts:106]: I don't think this belongs to math, throw it into collections.

**Review by stylemistake — APPROVED** [review]

**Comment by Rustybuckets6601** [comment]
You're still able to adjust suits right?

**Comment by Mothblocks** [comment]
Oops, knew I forgot something.

**Pull request merged**